Dockerfile

```diff
@@ -17,36 +17,34 @@ ENV EAP_VERSION_MICRO 0
-USER root
-RUN chown 1000:1000 /opt/jboss/$EAP_INSTALLER /opt/jboss/$BPMS_DEPLOYABLE
-
-# Prepare and run installer and cleanup installation components
-RUN sed -i "s:<installpath>.*</installpath>:<installpath>$BPMS_HOME</installpath>:" /opt/jboss/installation-eap \
-    && java -jar /opt/jboss/$EAP_INSTALLER  /opt/jboss/installation-eap -variablefile /opt/jboss/installation-eap.variables \
-    && unzip -qo /opt/jboss/$BPMS_DEPLOYABLE  -d $BPMS_HOME/.. \
-    && chown -R 1000:root $BPMS_HOME \
-    && /opt/jboss/fix-permissions $BPMS_HOME \
-    && rm -rf /opt/jboss/$BPMS_DEPLOYABLE /opt/jboss/$EAP_INSTALLER /opt/jboss/installation-eap /opt/jboss/installation-eap.variables $BPMS_HOME/standalone/configuration/standalone_xml_history/ \
-    && $BPMS_HOME/bin/add-user.sh -a -r ApplicationRealm -u erics -p bpmsuite1! -ro analyst,admin,manager,user,kie-server,kiemgmt,rest-all --silent
-
-# Add support files
-COPY support/standalone.xml $BPMS_HOME/standalone/configuration/
-COPY support/userinfo.properties $BPMS_HOME/standalone/deployments/business-central.war/WEB-INF/classes/
-
-# Fix permissions on support files
-RUN /opt/jboss/fix-permissions $BPMS_HOME/standalone/configuration/standalone.xml && \
-    /opt/jboss/fix-permissions $BPMS_HOME/standalone/deployments/business-central.war/WEB-INF/classes/userinfo.properties
+COPY support/installation-eap support/installation-eap.variables support/standalone.xml support/userinfo.properties installs/*.zip installs/*.jar /opt/jboss/
 
+#Update Permissions on Installers
 # Run as JBoss 
 USER 1000
 
+# Prepare and run installer and cleanup installation components
+RUN sed -i "s:<installpath>.*</installpath>:<installpath>$BPMS_HOME</installpath>:" /opt/jboss/installation-eap && \
+    java -jar /opt/jboss/$EAP_INSTALLER  /opt/jboss/installation-eap -variablefile /opt/jboss/installation-eap.variables && \ 
+    if [ -e /opt/jboss/$EAP_PATCH ]; then $BPMS_HOME/bin/jboss-cli.sh --commands=patch\ apply\ /opt/jboss/$EAP_PATCH\ --override-all; fi && \
+    unzip -qo /opt/jboss/$BPMS_DEPLOYABLE  -d $BPMS_HOME/.. && \
+    for f in /opt/jboss/$BPMS_PATCH_WILDCARD; do \
+      unzip -qo $f -d /opt/jboss/ && \
+      (cd ${f::-4} && exec ./apply-updates.sh $BPMS_HOME eap7.x); \
+    done && \
+    mv /opt/jboss/standalone.xml $BPMS_HOME/standalone/configuration/ && \
+    mv /opt/jboss/userinfo.properties $BPMS_HOME/standalone/deployments/business-central.war/WEB-INF/classes/ && \
+    rm -rf /opt/jboss/$BPMS_DEPLOYABLE /opt/jboss/$EAP_INSTALLER /opt/jboss/installation-eap /opt/jboss/installation-eap.variables $BPMS_HOME/standalone/configuration/standalone_xml_history/ && \
+    rm -rf /opt/jboss/$EAP_PATCH /opt/jboss/$BPMS_PATCH_WILDCARD && \
+    $BPMS_HOME/bin/add-user.sh -a -u bpmsAdmin -p BPMs3cr3t --role admin,developer,analyst,user,manager,kie-server,rest-all,Administrators
+
+# Extra users
+# RUN $BPMS_HOME/bin/add-user.sh -a -u user,casehandler1 -p BPMs3cr3t --role casehandler
```

Comment:
Won't be changing this either... if you need more roles, this can be edited in the file before running.

Reply:
You could add this because if there are no fixes in the directory it works fine.
In this way, it can be used even by subscribers to build a fixed demo env.